### PR TITLE
Rely on NodeListener instead of periodic detection

### PR DESCRIPTION
**Pull request by olivergondza**
https://issues.jenkins-ci.org/browse/JENKINS-33780

Can not be merged without rising the core version requirement drastically.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java [hunk 1]
```diff
@@ -475,6 +524,17 @@ protected void doRun() throws Exception {
         return ((TrackedItem) plannedNode).getId();
     }
 
+    private static @CheckForNull ProvisioningActivity.Id getIdFor(Node node) {
+        if (node instanceof Jenkins) return null;
+
+        if (!(node instanceof TrackedItem)) {
+            LOGGER.info("No support for cloud-stats-plugin by " + node.getClass());
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]: Btw, it spams system log.

Comment (olivergondza) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]: I guess so. We should do it once per type. Though, this behavior is not related to this PR, I presume that is how you found that out.

Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]: Were looking plugin status. Idea is useful, still thinking whether to use it or make own.

Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]: I think you can scan for existing ExtensionPoints and check inheritance. It could be done somewhere on last initialiasation steps when all plugins initialiased.

Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]:
```
Dec 21, 2016 5:32:31 PM org.jenkinsci.plugins.cloudstats.CloudStatistics getIdFor
INFO: No support for cloud-stats-plugin by class hudson.slaves.SlaveComputer
Dec 21, 2016 5:32:31 PM org.jenkinsci.plugins.cloudstats.CloudStatistics getIdFor
INFO: No support for cloud-stats-plugin by class hudson.slaves.SlaveComputer
Dec 21, 2016 5:45:31 PM hudson.slaves.SlaveComputer tryReconnect
INFO: Attempting to reconnect sdf
Dec 21, 2016 5:45:31 PM org.jenkinsci.plugins.cloudstats.CloudStatistics getIdFor
INFO: No support for cloud-stats-plugin by class hudson.slaves.SlaveComputer
Dec 21, 2016 5:45:31 PM org.jenkinsci.plugins.cloudstats.CloudStatistics getIdFor
INFO: No support for cloud-stats-plugin by class hudson.slaves.SlaveComputer
```
I guess it also spams about DumbSlave every N minutes.

Comment (olivergondza) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]: Addressed in https://github.com/jenkinsci/cloud-stats-plugin/commit/ba2f509e8147fd6322ae4b4b45fc2fff4b0742a6 and https://github.com/jenkinsci/cloud-stats-plugin/commit/e40d20af6b7d3bc3bdd6c898b6f116d008c5ab06.

**Comment by olivergondza** [comment]
Todo:
- [ ] test rename
- [ ] test detection

File: src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java (hunk 1 above)
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java]: ok, good. Will update my dev env where it spams while i'm debugging docker cloud :)

**Comment by KostyaSha** [comment]
- [x] release from wrong branch

**Comment by KostyaSha** [comment]
You need ban wrong version, make normal release and only then bump core requirement, either it will block non 2.8 usage.
